activerecord/lib/active_record/transactions.rb

```diff
@@ -338,7 +338,7 @@ def before_committed! # :nodoc:
     # Ensure that it is not called if the object was never persisted (failed create),
     # but call it after the commit of a destroyed object.
     def committed!(should_run_callbacks: true) #:nodoc:
-      if should_run_callbacks && destroyed? || persisted?
+      if should_run_callbacks && destroyed? || (persisted? && transaction_include_any_action?([:create, :update]))
```

Comment:
Compared to master, the logic here is a little off. There is a possibility `should_run_callbacks` gets passed as `false` and callbacks are still ran.

Reply:
Good point. That was just fixed on master 10 days ago, and I need to fix it in this branch too.